map_test.go

```diff
@@ -144,6 +144,61 @@ func createArray(t *testing.T) *Map {
 	return m
 }
 
+func TestMapQueue(t *testing.T) {
+
+	testutils.SkipOnOldKernel(t, "4.20", "map type queue")
+
+	m, err := NewMap(&MapSpec{
+		Type:       Queue,
+		ValueSize:  4,
+		MaxEntries: 2,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer m.Close()
+
+	if err := m.Put(nil, uint32(42)); err != nil {
+		t.Fatal("Can't put 42:", err)
+	}
+
+	if err := m.Put(nil, uint32(4242)); err != nil {
+		t.Fatal("Can't put 4242:", err)
+	}
+
+	var v uint32
+	if err := m.Lookup(nil, &v); err != nil {
+		t.Fatal("Can't lookup element:", err)
+	}
+	if v != 42 {
+		t.Error("Want value 42, got", v)
+	}
+
+	if err := m.LookupAndDelete(nil, &v); err != nil {
+		t.Fatal("Can't lookup and delete element:", err)
+	}
+	if v != 42 {
+		t.Error("Want value 42, got", v)
+	}
+
+	if err := m.Lookup(nil, &v); err != nil {
+		t.Fatal("Can't lookup element:", err)
+	}
+	if v != 4242 {
+		t.Error("Want value 4242, got", v)
+	}
+	if err := m.LookupAndDelete(nil, &v); err != nil {
```

Comment:
You need to reset v here.

```suggestion
	v = 0
	if err := m.LookupAndDelete(nil, &v); err != nil {
```